### PR TITLE
344 ML model and CLI improvements

### PR DESCRIPTION
**Pull request by nmaarnio**
This PR concerns both CLI and ML tools.

### CLI
- Added enum getter/unpacker (was required for list enums to be interpreted correctly but applied to all to be sure)
- All CLI functions with enum parameters use the above getter
- Harmonized how enum parameters are defined for CLI functions (now always `Annotated[EnumParameterType, typer.Option(case_sensitive=False) = EnumParameterType.some_default_value`)

### ML modeling
- Added links to Sklearn and Keras docs where relevant
- Added `criterion` parameter to Random Forest models
- Added raster grid check to ML model data preparation and renamed training rasters -> feature rasters for clarity (since some call labels as training data)

### CODE REVIEW

**Review by RichardScottOZ — COMMENTED** [review]
Yes, sensible renaming.

**Review by Mtk112 — APPROVED** [review]
Looks good to me!
